### PR TITLE
Feat: Added budget back end functionality

### PR DESCRIPTION
**Pull request by Jmartimus**
In this PR, I reorganized the back end data structure files (interfaces, DTOs, Schema).  I wrote the functionality for creating/gettingall/editing budgets.  I also created a ```const months``` array to use for a dropdown on the front end.

### CODE REVIEW

File: client/src/constants.ts [hunk 1]
```diff
@@ -23,3 +23,18 @@ export interface ItemData {
...
+  "April",
+  "May",
+  "June",
+  "July",
+  "August",
+  "September",
+  "October",
+  "November",
+  "December",
+] as const;
```
Comment (bycho91) [inline client/src/constants.ts:40]: 'as const'? Maybe typo?

**Review by bycho91 — CHANGES_REQUESTED** [review]
Wasn't sure if this was a typo or intentional

File: client/src/constants.ts (hunk 1 above)
Comment (Jmartimus) [reply client/src/constants.ts:40]: Not a typo, in JavaScript, const actually can be changed in some ways. With typescript you can use "as const" to make it a true Const that cannot be changed.

Also, I forgot that it should be in all caps. MONTHS. It's just to emphasize that it is a true const.  Maybe you can make that change when you start working with it.

Comment (bycho91) [reply client/src/constants.ts:40]: ahhh Never seen this before! Good to know!

**Review by bycho91 — APPROVED** [review]
Woot

**Pull request merged**